jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py

```diff
@@ -278,6 +283,11 @@ def _on_add_data_to_viewer(self, msg):
             self._handle_attribute_display_unit(self.sb_unit_selected, layers=layers)
             self._clear_cache('image_layers')
 
+    def _on_slice_changed(self, msg):
+        if self.config != "cubeviz":
+            return
+        self._cube_wave = u.Quantity(msg.value, msg.value_unit)
```

Comment:
when this changes, don't we need to re-convert anything that is not in native units?  Surely this is then needed in other places, should we generalize it into a mixin?

Reply:
I am not familiar enough with the unit conversion machinery here to generalize it as part of this PR.